associate.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import os
+
+
+def read_file_list(filename, isRGB):
+    file = open(filename)
+    data = file.read()
+    lines = data.replace(","," ").replace("\t"," ").split("\n")
+    lines = [line for line in lines if line != '']
+    lines = list(map(float, lines))
+    files = None
+    if isRGB:
+        files = os.listdir(os.path.join(os.path.dirname(filename), "color"))
+    else:
+        files = os.listdir(os.path.join(os.path.dirname(filename), "depth"))
+    if isRGB:
+        files = [os.path.join("color", x) for x in files]
+    else:
+        files = [os.path.join("depth", x) for x in files]
+    files.sort()
+    d = dict(zip(lines, files))
+    return d
+
+
+def associate(first_list, second_list,offset,max_difference):
+    first_keys = list(first_list.keys())
+    second_keys = list(second_list.keys())
+    potential_matches = [(abs(a - (b + offset)), a, b) 
+                         for a in first_keys 
+                         for b in second_keys 
+                         if abs(a - (b + offset)) < max_difference]
+    potential_matches.sort()
+    matches = []
+    for diff, a, b in potential_matches:
+        if a in first_keys and b in second_keys:
+            first_keys.remove(a)
+            second_keys.remove(b)
+            matches.append((a, b))
+    
+    matches.sort()
+    return matches
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='''
+    This script takes two data files with timestamps and associates them   
+    ''')
+    parser.add_argument('first_file', help='first text file (format: timestamp data)')
```

Comment:
Why do you need timestamp files? You can use only paths to rgb and depth folders and match them. Naming of images is self-contained.